src/Apps/Order/Components/OfferInput.tsx

```diff
@@ -20,28 +20,28 @@ export const OfferInput: FC<OfferInputProps> = ({
   onChange,
   value,
 }) => {
+  const formatValueForDisplay = (val: number | undefined) => {
+    if (val !== undefined) {
+      return val.toLocaleString()
+    }
+    return ""
+  }
+
   return (
     <Input
       id={id}
       title={noTitle ? "" : "Your offer"}
       type="text"
       pattern="[0-9]"
       error={showError ? "Offer amount missing or invalid." : false}
+      inputMode={"numeric"}
       onFocus={onFocus}
       onBlur={onBlur}
-      value={value}
+      value={formatValueForDisplay(value)}
       onChange={ev => {
         const currentValue = ev.currentTarget.value
-        const nonDigitMatch = currentValue.match(/\D/)
-
-        if (nonDigitMatch) {
-          const cursorOffset = currentValue.indexOf(nonDigitMatch[0])
-          const nextValue = currentValue.replace(/\D/g, "")
-          ev.currentTarget.value = nextValue
-          ev.currentTarget.setSelectionRange(cursorOffset, cursorOffset)
-        }
-
-        onChange(Number(ev.currentTarget.value || "0"))
+        const cleanedValue = currentValue.replace(/[^\d]/g, "") // Remove non-digits
+        onChange(Number(cleanedValue))
```

Comment:
I think that is an expected behaviour depending on the design path the platform goes. I can reproduce this "erase to 0" pattern in Paypal and N26, for example. Since we got design approval for this approach just now, we can try this out in staging mobile/web and see how it goes :)

Reply:
Sounds good 👍